src/util-mpm-hs.c

```diff
@@ -570,174 +622,475 @@ static PatternDatabase *PatternDatabaseAlloc(uint32_t pattern_cnt)
     return pd;
 }
 
+static const char *HSCacheConstructFPath(uint64_t hs_db_hash)
+{
+    static char hash_file_path[PATH_MAX];
+
+    char hash_file_path_suffix[] = "_v1.hs";
+    char filename[PATH_MAX];
+    int r = snprintf(filename, sizeof(filename), "%020lu%s", hs_db_hash, hash_file_path_suffix);
+    if (r != (int)(20 + strlen(hash_file_path_suffix)))
+        return NULL;
+
+    r = PathMerge(
+            hash_file_path, sizeof(hash_file_path), DetectEngineMpmCachingGetPath(), filename);
```

Comment:
what would it take to avoid the global `DetectEngineMpmCachingGetPath()` but instead pass a path along?

Reply:
quite a bit actually. We don't pass anything from the outside (meaning Suricata codebase) to MPM matches except AddPattern function. I could:
- leave it as is
- initialize the variable mpm_ctx in SCHSInitCtx() - but I believe it would work a similar way as it is now
- redo MPM API - add a possibility to pass configuration to the MPM matcher before anything gets initialized there.

I would suggest to stick with the current implementation as I don't see why we would want to dynamically change the configuration path. If you see a reason for that then I would go with the third option.